### ISSUE TITLE
Gallery block: images selection and removal not keyboard accessible

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#3836]
In the Gallery block, clicking on an image makes the "remove image" button appear:

![screen shot 2017-12-06 at 21 26 10](https://user-images.githubusercontent.com/1682452/33683949-de845872-dacc-11e7-9307-b5c8caef89c3.png)

However, when using only the keyboard, the images can't be selected nor the remove button appears (it's not rendered by default).

Any functionality should be operable also when using only the keyboard. Instead of making the images clickable, I'd recommend to use a separate control always rendered in the markup that becomes visible on focus. This control should be properly labeled and have some reference to the image that will be removed.

Aside: when an image is "visually selected", it can't be de-selected. I wonder if the selection is really necessary, as the only available action seems to be removing the images one by one.

**Comment by afercia** [comment]
Reporting from #3790:

Worth noting the gallery block, when an image is selected, displays a "remove image" button that uses `$blue-medium-500` as background and a white "X" as text: this contrast ratio is not sufficient and should be changed. I guess it would need some design feedback.

![screen shot 2017-12-03 at 15 03 50](https://user-images.githubusercontent.com/1682452/33530138-75f2d060-d840-11e7-899b-0542c6fb15e2.png)

**Comment by afercia** [comment]
An example of existing pattern in the Customizer;

<img width="300" alt="screen shot 2017-12-10 at 12 04 37" src="https://user-images.githubusercontent.com/1682452/33810483-51c41c56-de05-11e7-9d47-46cfb940c777.png">

**Comment by SuperGeniusZeb** [comment]
I think this should be added to the Merge Proposal: Accessibility milestone.

**Comment by bemdesign-er** [comment]
The gallery block needs a way for keyboard users to navigate between images inside the gallery block, a way to make selections images, and a way to delete those selected images from the gallery block.

**Comment by afercia** [comment]
Discussed during today's accessibility bug scrub. Yep, still an issue. Adding the "bug" label as the gallery block is still not usable with a keyboard.

**Comment by joedolson** [comment]
Still an issue. From what I can tell, this requires that the images be made focusable. If the images are focused using the mouse, you can navigate the controls for them using the keyboard; but there's no apparent way to focus the images without a mouse.
